src/interface.ts

```diff
@@ -25,10 +43,19 @@ interface AppMetadata {
 
 /**
  * IntentResolution provides a standard format for data returned upon resolving an intent.
+ * ```javascript
+ * //resolve a "Chain" type intent
+ * var intentR = await agent.raiseIntent("intentName", context);
+ * //resolve a "Client-Service" type intent with data response
+ * var intentR = await agent.raiseIntent("intentName", context);
+ * if (intentR.data){
+ *  var dataR = await intentR.data();
+ * }
+ * ```
  */
 interface IntentResolution {
   source: String;
-  data?: Object; 
+  data?: ()=> Promise<Object>; 
```

Comment:
I am tempted to ask if we can do this part (which tries to address #31) as a separate PR, because I would still like to discuss the way that we do this, and I think the other changes are ready to merge.

Reply:
np.  I'll back this out and merge